packages/protocol/test/common/integration.ts

```diff
@@ -559,3 +559,153 @@ contract('Integration: Exchange', (accounts: string[]) => {
+  // passing a governance proposal for each one will be a pain in the butt.
+  before(async () => {
+    const proposalId = 2
+    const dequeuedIndex = 0
+    const lockedGold = await getDeployedProxiedContract('LockedGold', artifacts)
+    // @ts-ignore
+    await lockedGold.lock({ value: '10000000000000000000000000' })
+    const value = await lockedGold.getAccountTotalLockedGold(accounts[0])
+    const multiSig = await getDeployedProxiedContract('GovernanceApproverMultiSig', artifacts)
+    const governance = await getDeployedProxiedContract('Governance', artifacts)
+    const registry = await getDeployedProxiedContract('Registry', artifacts)
+    const contractsToOwn = ['Freezer', 'Registry', 'Reserve', 'SortedOracles']
+    const proposalTransactions = await Promise.all(
+      contractsToOwn.map(async (x: string) => {
+        return {
+          value: 0,
+          destination: (await getDeployedProxiedContract(x, artifacts)).address,
+          data: Buffer.from(
+            stripHexEncoding(
+              // Any contract's `setOwner` function will work here as the function signatures are all the same.
+              // @ts-ignore
+              registry.contract.methods.setOwner(accounts[0]).encodeABI()
+            ),
+            'hex'
+          ),
+        }
+      })
+    )
+    await governance.propose(
+      proposalTransactions.map((x: any) => x.value),
+      proposalTransactions.map((x: any) => x.destination),
+      // @ts-ignore
+      Buffer.concat(proposalTransactions.map((x: any) => x.data)),
+      proposalTransactions.map((x: any) => x.data.length),
+      'URL',
+      // @ts-ignore: TODO(mcortesi) fix typings for TransactionDetails
+      { value: web3.utils.toWei(config.governance.minDeposit.toString(), 'ether') }
+    )
+
+    await governance.upvote(proposalId, 0, 0)
+    await timeTravel(config.governance.dequeueFrequency, web3)
+    // @ts-ignore
+    const txData = governance.contract.methods.approve(proposalId, dequeuedIndex).encodeABI()
+    await multiSig.submitTransaction(governance.address, 0, txData, {
+      from: accounts[0],
+    })
+    await timeTravel(config.governance.approvalStageDuration, web3)
+    await governance.vote(proposalId, dequeuedIndex, VoteValue.Yes)
+    await timeTravel(config.governance.referendumStageDuration, web3)
+    await governance.execute(proposalId, dequeuedIndex)
+  })
+
+  // 1. Mimic the state of the world post-contracts-release
+  //   a) Deploy the contracts. For simplicity, omit proxies for now.
+  //   b) Initialize and register the contracts
+  //   c) Confirm mento is effectively frozen
+  describe('When the contracts have been deployed and initialized', () => {
+    before(async () => {
+      exchangeABC = await Exchange.new()
+      stableTokenABC = await StableToken.new()
+
+      // TODO: This is not very readable, what are these parameters and why were these values chosen?
+      // TODO: Which happens first, initialization or registration?
```

Comment:
looks like we register before initializing
https://github.com/celo-org/celo-monorepo/blob/05ca067f6152302842af190f7a2dc3a84d63a48d/packages/protocol/scripts/truffle/make-release.ts#L154

Reply:
registration isn't done here?

Reply:
Good catch